### DISCUSSION TITLE
Hyde variables

### DISCUSSION THREAD

**Discussion by dieBakterie** [prasanthrangan/hyprdots#1530] [Q&A] [answered]
What's the userprefsdir variable for?

**Comment by kRHYME7** [comment]
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/656009c8-2cd2-4965-8f0b-1da7c91404a9)



![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/de23ad74-875c-4e3f-8a90-9c042595a923)

**Reply by dieBakterie** [reply]
That's for people who want their own dots with tweaks added but don't want to maintain an fork? Sorry if i don't get it,  but is there really a big difference between syncing from the cloned git repo and from a local folder? Is it a bad idea to symlink the cloned repo to your `~.config/`, `~/.local/share/bin`? Sorry If that's Off topic again.

**Reply by kRHYME7** [reply]
> That's for people who want their own dots with tweaks added but don't want to maintain an fork

Something like that,
* I only got this idea from Discord that they just have a backup of their preferef files, then they make a script to populate it back to the configs. It sounds useful for some. So I added the idea to the CLI. 
*  I am using it personally for files that I backed up, and I don't want to be overwritten. For example, modules in waybar etc... If  my waybar PR is merged I think this feature is not that useful. 


>  Is it a bad idea to symlink the cloned repo to your ~.config/, ~/.local/share/bin? Sorry If that's Off topic again.

Not really a bad Idea, not until that config is touch by unknown factors. It can be overwritten or something. It would sound like how chezmoi does things. 
What might happen is the symlink in the repo is also change, so you got a dirty git workspace.

**Reply by dieBakterie** [reply]
Everytime i do Changes i've to copy them over to the repo or vice versa. Chezmoi also has a third "version" between the local and the Git repo and idk how chezmoi works/handles forks. Should take a look at Chezmoi and i also do tweak the Scripts but i'm on my own fork.

**Reply by kRHYME7** [reply, marked as answer]
That's good if you fork it. You'll learn a lot, too.  
We can't really make a user preferred things if we just shared dotfiles. Because ~/.configs and anything in $HOME is userprefs. For now, the only 'real' userprefered variables are the ones in the `  ./hyde.conf `   other things is on your own. 

It's funny how we try to make  `dotfiles` of the ` dotfiles ` of the ` dotfiles ` . You get what I mean? 
So forking is the only way to handle dotfiles that is truly your own, TBH.

> [!Note]
>  Desktop environments mostly saves the the userprefs inside ~/.config

**Reply by dieBakterie** [reply]
Yeah i know but can't make .config a Git repo , don't want to make `hypr`, `hyprlock`, `Code/Code-OSS`, etc an Git repo, could do this but hell nah when i really know what i'm doing i'll propably do this xd
